.github/workflows/commit-lint.yml

```
@@ -8,7 +8,6 @@ jobs:
     steps:
       - uses: actions/checkout@v2
         with:
-          ref: ${{ github.event.pull_request.head.sha }}
           # Last 100 commits should be enough for a PR
           fetch-depth: 100
```

By default, all the commits are squashed into one, you need to opt-out of that (see https://github.com/actions/checkout)
```suggestion
          clean: false
          fetch-depth: ${{ github.event.pull_request.commits }} + 1
```

---

I've opened https://github.com/nodejs/node/pull/39123 to provide a more robust solution (at least I hope so) which is to always download the `problem-matcher.json` from nodejs/node/HEAD, PTAL.

---

> By default, all the commits are squashed into one, you need to opt-out of that (see actions/checkout#usage)

I don't see any references to it on actions/checkout#usage, could you point to the specific part of the documentation that asserts commits will be squashed?

> clean: false

```
    # Whether to execute `git clean -ffdx && git reset --hard HEAD` before fetching
    # Default: true
    clean: ''
```

Cleaning the workspace is a good idea (as it removes any unwanted files that might be there, even thought there shouldn't be anything there on the first step of the Action), I don't understand the benefit of setting `clean: false` here.

---

`${{ github.event.pull_request.commits }}` will return the API location of the Pull Request Commits ([github.event](https://docs.github.com/en/actions/reference/context-and-expression-syntax-for-github-actions#github-context) is webhook payload, for `pull_request` event that payload is defined [here](https://docs.github.com/en/developers/webhooks-and-events/webhooks/webhook-events-and-payloads#pull_request), so `github.event.pull_request` is defined [here](https://docs.github.com/en/rest/reference/pulls) and according to these docs `github.event.pull_request` will be a URL, not a number). 100 commits in a PR is so unlikely that using 100 as depth should be enough, and it's still small enough for `clone` to not take a long time.

---

> I don't see any references to it on actions/checkout#usage, could you point to the specific part of the documentation that asserts commits will be squashed?

I can't find it in their documentation, but I know it by experience. Here's a POC: https://github.com/aduh95/node-auto-test/pull/1/checks?check_run_id=2888719930. If you don't specify a `ref`, it will create a merge commit.

---

It created a merge commit, which is expected. No squashing. Because we use `git log --oneline ${{ github.event.pull_request.base.sha }}..${{ github.event.pull_request.head.sha }}` (as in, we run `git log` specifically for the range between the base and head of the pull request), the merge commit is not linted and therefore this Action will still work.

---

`${{ github.event.pull_request.commits }}` is a number, and correspond to the number of commits in a given PR: https://github.com/aduh95/node-auto-test/runs/2888781556
You may want to add `+ 1` to avoid fetching everything for PR with 0 commits.

---

Interesting, I'm generally concerned about using undocumented features (but not opposed to it). You sure the `+ 1` operand also works (as in the Actions runner will perform an operation before passing it to `checkout`, instead of generating a "number_of_commits + 1" string? (technically this could be a separate PR since it's not necessary to fix the linter action, but I don't mind including it here)

w.r.t `clean`, what's the intention behind it?

---

> w.r.t `clean`, what's the intention behind it?

I misread the documentation, I thought it was doing a squashing. Please disregard.

> You sure the `+ 1` operand also works (as in the Actions runner will perform an operation before passing it to `checkout`, instead of generating a "number_of_commits + 1" string?

You're right, it doesn't work indeed: https://github.com/aduh95/node-auto-test/pull/1/checks?check_run_id=2888913860. A better solution would be to skip the test altogether if the number of commits is 0. I agree it deserves its own PR, I'll make one momentarily.

---

Actions does have operators, but the documentation is [bad](https://docs.github.com/en/actions/reference/context-and-expression-syntax-for-github-actions#operators). I should've clarified on my previous comment: I wasn't sure if the +1 should be inside or outside ${{ }} or something else. The +1 idea should work, I just don't remember the correct syntax 😅